x/mongo/driver/connstring/connstring_test.go

```diff
@@ -102,6 +102,11 @@ func TestConnect(t *testing.T) {
 		{s: "connect=AUTOMATIC", expected: connstring.AutoConnect},
 		{s: "connect=direct", expected: connstring.SingleConnect},
 		{s: "connect=blah", err: true},
+		// combinations of connect and directConnection - conflicting combinations must error
+		{s: "connect=automatic&directConnection=true", err: true},
```

Comment:
What about the case where directConnections is already set and then connect gets set?

Reply:
Added the same four tests in reverse order below.